src/Microsoft.ML.StaticPipe/FactorizationMachineStatic.cs

```diff
@@ -24,46 +24,60 @@ public static class FactorizationMachineExtensions
         /// be informed about what was learnt.</param>
         /// <returns>The predicted output.</returns>
         public static (Scalar<float> score, Scalar<bool> predictedLabel) FieldAwareFactorizationMachine(this BinaryClassificationCatalog.BinaryClassificationTrainers catalog,
             Scalar<bool> label, Vector<float>[] features,
-            float learningRate = 0.1f,
-            int numIterations = 5,
-            int numLatentDimensions = 20,
-            Action<FieldAwareFactorizationMachineTrainer.Arguments> advancedSettings = null,
             Action<FieldAwareFactorizationMachineModelParameters> onFit = null)
         {
             Contracts.CheckValue(label, nameof(label));
             Contracts.CheckNonEmpty(features, nameof(features));
 
-            Contracts.CheckParam(learningRate > 0, nameof(learningRate), "Must be positive");
-            Contracts.CheckParam(numIterations > 0, nameof(numIterations), "Must be positive");
-            Contracts.CheckParam(numLatentDimensions > 0, nameof(numLatentDimensions), "Must be positive");
-            Contracts.CheckValueOrNull(advancedSettings);
             Contracts.CheckValueOrNull(onFit);
 
             var rec = new CustomReconciler((env, labelCol, featureCols) =>
             {
-                var trainer = new FieldAwareFactorizationMachineTrainer(env, featureCols, labelCol, advancedSettings:
-                    args =>
-                    {
-                        args.LearningRate = learningRate;
-                        args.Iters = numIterations;
-                        args.LatentDim = numLatentDimensions;
-
-                        advancedSettings?.Invoke(args);
-                    });
+                var trainer = new FieldAwareFactorizationMachineTrainer(env, featureCols, labelCol);
+
+                if (onFit != null)
+                    return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                else
+                    return trainer;
+            }, label, features);
+            return rec.Output;
+        }
+
+        /// <summary>
+        /// Predict a target using a field-aware factorization machine.
+        /// </summary>
+        /// <param name="catalog">The binary classifier catalog trainer object.</param>
+        /// <param name="label">The label, or dependent variable.</param>
+        /// <param name="features">The features, or independent variables.</param>
+        /// <param name="options">Advanced arguments to the algorithm.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the model that was trained.  Note that this action cannot change the result in any way; it is only a way for the caller to
```

Comment:
What's the type of `the model that was trained`? Is it possible to make a reference here?